### DISCUSSION TITLE
How to get fa decimals inside the move contract from metadata??

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#377] [Questions] [answered]
is there any function to fetch decimals of fa metadata

**Comment by MoonShiesty** [comment, marked as answer]
`public fun decimals<T: key>(metadata: Object<T>)`

**Reply by Alexzander-Stone** [reply]
The above is correct! 
I'd like to add on that the [Move Reference docs](https://legacy.aptos.dev/reference/move) are a great place to learn about functionality of various modules (like [Fungible Assets](https://legacy.aptos.dev/reference/move/?branch=mainnet&page=aptos-framework/doc/fungible_asset.md#0x1_fungible_asset_decimals)), @ajaythxkur.

**Reply by ajaythxkur** [reply]
Thanks
